src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,234 @@
+        public OperatorIndexerCompletionProvider()
+        {
+        }
+
+        private const string MinimalTypeNamePropertyName = "MinimalTypeName";
+        private const string CompletionHandlerPropertyName = "CompletionHandler";
+        private const string CompletionHandlerConversion = "Conversion";
+        private const string CompletionHandlerIndexer = "Indexer";
+
+        // Identifier can start with "A Unicode character of classes Lu, Ll, Lt, Lm, Lo, or Nl". https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/lexical-structure#identifiers
+        // Sorting is done via StringComparer.OrdinalIgnoreCase which compares the utf-16 bytes after converting to uppercase
+        // utf-16 \ufffd is the largest possible value for utf-16 and is also greater than surrogate pairs, if byte comparison is used.
+        // The "biggest" possible characters are \u3134a http://www.fileformat.info/info/unicode/char/3134a/index.htm surrogate pair "\ud884\udf4a" and
+        // \uffdc http://www.fileformat.info/info/unicode/char/ffdc/index.htm (non-surrogate)
+        private const string SortingPrefix = "\uFFFD";
+
+        internal override ImmutableHashSet<char> TriggerCharacters => ImmutableHashSet.Create('.');
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '.';
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        private static ImmutableDictionary<string, string> CreateCompletionHandlerProperty(string operation, params (string key, string value)[] additionalProperties)
+        {
+            var builder = ImmutableDictionary.CreateBuilder<string, string>();
+            builder.Add(CompletionHandlerPropertyName, operation);
+            foreach (var (key, value) in additionalProperties)
+            {
+                builder.Add(key, value);
+            }
+
+            return builder.ToImmutable();
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var position = context.Position;
+            var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            token = token.GetPreviousTokenIfTouchingWord(position);
+            if (!token.IsKind(SyntaxKind.DotToken))
+            {
+                return;
+            }
+
+            var expression = GetParentExpressionOfInvocation(token);
```

Comment:
i don't relaly understand the method name in this context.  it says "getparent ... of invocation", but what makes us think we have an invocationhere?